docs/source/extensions.rst

```diff
@@ -216,8 +216,45 @@ elements in a dataset.
 The _custom dictionary
 ^^^^^^^^^^^^^^^^^^^^^^
 
-TODO (in progress ...)
+All cfdm classes have a `_custom` attribute that contains a dictionary
+meant for use in external subclasses. It is not used by any cfdm
+classes. The `_custom` dictionary is shallow copied, rather than deep
+copied, when using the standard cfdm deep copy method techniques
+(i.e. the `!copy` method, initialisation with the *source* parameter,
+or applying the `copy.deepcopy` function). This is so that subclasses
+of cfdm are not committed to potentially expensive deep copies of the
+dictionary values. Note that calling `copy.deepcopy` on cfdm
+(sub)class simply invokes its `!copy` method.
+
+The result of this behaviour is that if an external subclass stores a
+mutable object within its custom dictionary then, by default, a deep
+copy will contain the identical mutable object, to which in-place
+changes will affect both the original and copied instances.
+
+To account for this, the external subclass can either simply commit to
+never updating such mutables in-place (which is can be acceptable for
+private quantities which are tightly controlled); or else include
+extra code that does deep copy such mutables when any deep copy (or
+equivalent) operation is called. The latter should be implemented in
+the subclasses `__init__` method, similarly to this:
 
+.. code-block:: python
+   :caption: *Ensure that the _custom dictionary 'x' value is deep
+             copied when a deep copy of an instance is requested.*
+	 
+   >>> class my_Field_3(cfdm.Field):
```

Comment:
In the case of this code example, and perhaps some of the others on the 'extension' page where they are no outputs to show, could it be clearer to show it as a pure Python block rather than in iPython with the prompts and need for continuation characters afterwards?

Reply:
Good idea - I'll sort this out on Monday ...

Reply:
Please do, it is solidly into weekend time now :grinning: Have a good one.